### PR TITLE
[LinkerWrapper] Fix `-Xoffload-linker a=b` in offloading

### PR DESCRIPTION
**Pull request by jhuber6**
Summary:
We have the `-Xoffload-linker=triple=arg` syntax that split the argument
meant only for a single toolchain. However this borke if it was an `a=b`
type argument. Make it only treat it like a triple if it's a valid
triple.

### CODE REVIEW

**Review by shiltian — COMMENTED** [review]
I think LLVM generally doesn't follow the convention that `-a x` and `--abc=x`?

**Comment by jhuber6** [comment]
> I think LLVM generally doesn't follow the convention that `-a x` and `--abc=x`?

`cl::opt` options are arbitrary, but this isn't about convention really, it's for things like `-Xoffload-linker -Wl,-plugin-opt=blah` not being passed correctly.

File: clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp [hunk 1]
```diff
@@ -1214,7 +1214,13 @@ DerivedArgList getLinkerArgs(ArrayRef<OffloadFile> Input,
   // Forward '-Xoffload-linker' options to the appropriate backend.
   for (StringRef Arg : Args.getAllArgValues(OPT_device_linker_args_EQ)) {
     auto [Triple, Value] = Arg.split('=');
-    if (Value.empty())
+    llvm::Triple TT(Triple);
+    // If this isn't a recognized triple then it's an `arg=value` option.
+    if (TT.getArch() <= Triple::ArchType::UnknownArch ||
+        TT.getArch() >= Triple::ArchType::LastArchType)
```
Comment (shiltian) [inline clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp:1220]: should it be `TT.getArch() >= Triple::ArchType::LastArchType` as the last arch type is valid, which is `ve`?

Comment (jhuber6) [reply clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp:1220]: Good catch
```suggestion
        TT.getArch() > Triple::ArchType::LastArchType)
```

**Review by shiltian — APPROVED** [review]

**Pull request merged**